internal/db/advisory_locks_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2025 Canonical.
+
+package db_test
+
+import (
+	"testing"
+	"time"
+
+	qt "github.com/frankban/quicktest"
+	"github.com/frankban/quicktest/qtsuite"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+
+	"github.com/canonical/jimm/v3/internal/db"
+	"github.com/canonical/jimm/v3/internal/testutils/jimmtest"
+)
+
+type advisoryLocksSuite struct{}
+
+func (s *advisoryLocksSuite) TestAdvisory_LockAndUnlock(c *qt.C) {
+	ctx := c.Context()
+
+	gdb1, _ := jimmtest.PostgresDBWithDbName(c, time.Now)
+
+	db1 := &db.Database{
+		DB: gdb1,
+	}
+
+	sqldb, err := db1.DB.DB()
+	c.Assert(err, qt.IsNil, qt.Commentf("Failed to get SQL DB connection"))
+	sqlconn, err := sqldb.Conn(ctx)
+	c.Assert(err, qt.IsNil, qt.Commentf("Failed to get SQL connection"))
+	gdb2, err := gorm.Open(postgres.New(postgres.Config{
+		Conn: sqlconn,
+	}))
+	c.Assert(err, qt.IsNil, qt.Commentf("Failed to open second GORM DB connection"))
+
+	db2 := &db.Database{
+		DB: gdb2,
+	}
+
+	// Acquire lock in db session 1.
+	err = db1.LockBootstrap(ctx)
+	c.Assert(err, qt.IsNil, qt.Commentf("Failed to acquire lock"))
+
+	// Attempt to acquire lock in session 2, should fail.
+	err = db2.LockBootstrap(ctx)
+	c.Assert(
```

Comment:
Feel free to ignore this one, but without a restriction on line length in jimm, I wouldn't mind if you kept this as a single line to match the other assertions.